signac/core/zarrcollection.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Implements Zarr-backend.
+
+This implements the Zarr-backend for SyncedCollection API by
+implementing sync and load methods.
+"""
+from .synced_collection import SyncedCollection
+from .syncedattrdict import SyncedAttrDict
+from .synced_list import SyncedList
+
+
+class ZarrCollection(SyncedCollection):
+    """Implement sync and load using a Zarr backend."""
+
+    backend = __name__  # type: ignore
+
+    def __init__(self, name=None, store=None, **kwargs):
+        import zarr
+        import numcodecs  # zarr depends on numcodecs
+
+        self._root = zarr.group(store=store)
+        self._name = name
+        self._object_codec = numcodecs.JSON()
+        super().__init__(**kwargs)
+        if (name is None) == (self._parent is None):
+            raise ValueError(
+                "Illegal argument combination, one of the two arguments, "
+                "parent or name must be None, but not both.")
```

Comment:
We keep repeating this particular check in all constructors, maybe we can move that into the parent constructor?